frontend/src/components/core/LockScreen.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { InlineShowIcon } from './Button';
+import ErrorMessage from './Error';
+import { LargeCenterPassword, PrimaryInput } from './Input';
+import Loading from './Loading';
+import { LargeText, SmallHoverText } from './Text';
+
+const Content = styled.div`
+  padding: 0 20%;
+`;
+
+type LockScreenProps = {
+  loading: boolean,
+  error: string,
+  enterPasswordAction: (password: string) => void,
+};
+
+export default function LockScreen(props: LockScreenProps) {
+  // Grab props variables.
+  const {
+    error, loading, enterPasswordAction,
+  } = props;
+
+  const [password, setPassword] = useState('');
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  const togglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  return (
+    <Content>
+      <LargeText>
+        What is the password?
+      </LargeText>
+      <LargeCenterPassword
+        show={showPassword}
+        placeholder="Password"
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          setPassword(event.target.value);
+        }}
+        onKeyPress={(event: React.KeyboardEvent<HTMLInputElement>) => {
+          if (event.key === 'Enter') {
+            enterPasswordAction(password);
+          }
+        }}
+      />
+      <SmallHoverText
+        onClick={togglePasswordVisibility}
+      >
+        {showPassword ? 'Hide' : 'Show'}
```

Comment:
I did try doing that, but couldn't figure out how to do that easily as the `<input>` elements cannot hold any children. I can do this by positioning the eye element absolutely from a parent container, but I was also worried about the "show / hide" icon overlapping on the text. I can add additional padding around the text and add the "eye" icon after, but if I put it in the same place as the "LastPass" element, no matter what z-index I use it is always positioned below (see image).

<img width="360" alt="Show / Hide Icon on Input" src="https://user-images.githubusercontent.com/7987279/113518386-dc347800-953a-11eb-9878-4b4931f4f2c7.png">

To avoid conflict with the Chrome extensions and whatnot that might use that space, and just for simplicity's sake, I think I'd just prefer the simpler solution of just having clickable text below the input. However, I'm still open to implementing this in some form, just wanted to share my reasoning - thoughts?

Reply:
Awesome thanks! Yeah the LastPass thing is I guess a little problematic haha - is it still relatively clickable? Either way I think this looks good and am okay with it (but if you find that overlap too annoying feel free to change it back, maybe made smaller or aligned left/closer to the input box tho)

Reply:
Note: I don't think this change has been pushed yet? It's not showing up for me locally

Reply:
Yeah so I never actually implemented this, I just updated the frontend through inspect element (that's why I labeled it as just an "example"). The "eye" button is relatively clickable, but since I think a lot of extensions use that space and I have some concerns about the styling / z-index, I might just leave the text below. I'll make those changes you mentioned though! (Smaller text size and whatnot.)

Reply:
Done! ✅  Image of the implementation below - open to changing this in future PRs!

<img width="1268" alt="Implement Show / Hide Button Design" src="https://user-images.githubusercontent.com/7987279/113563980-7210e700-95bd-11eb-93e8-78cc4bc2d8fb.png">